magda-authorization-api/src/test/NestedSetModelQueryer.spec.ts

```diff
@@ -0,0 +1,950 @@
+        pool = new pg.Pool({ ...dbConfig });
+        try {
+            await pool.query("CREATE database test");
+        } catch (e) {
+            // --- if database `test` already there
+            // --- then mute the error
+            if (e.code !== "42P04") {
+                throw e;
+            }
+        }
+        // --- end the current one & create a new one
+        await pool.end();
+        pool = new pg.Pool({ ...dbConfig, database: "test" });
+        await pool.query('CREATE EXTENSION IF NOT EXISTS "uuid-ossp"');
+    });
+
+    after(async function() {
+        if (pool) {
+            for (let i = 0; i < createTables.length; i++) {
+                await pool.query(`DROP TABLE IF EXISTS "${createTables[i]}"`);
+            }
+            pool.end();
+            pool = null;
+        }
+    });
+
+    async function createTestTable() {
+        const tableName = `test_tbl_${Math.random()
+            .toString()
+            .replace(".", "")}`;
+        await pool.query(`CREATE TABLE "${tableName}" (
+            "id" uuid NOT NULL DEFAULT uuid_generate_v4(),
+            "name" varchar(250) NOT NULL DEFAULT ''::character varying,
+            "desc" varchar(250) NOT NULL DEFAULT ''::character varying,
+            "left" int4 NOT NULL,
+            "right" int4 NOT NULL,
+            PRIMARY KEY ("id")
+        ) WITH (
+            OIDS = FALSE
+        )`);
+        createTables.push(tableName);
+        return tableName;
+    }
+
+    /**
+     * Create test data as followings:
+     *
+     *         +---------+
+     *         |  Albert |
+     *         | 1     12|
+     *         +----+----+
+     *              |
+     *     +--------+--------+
+     *     |                 |
+     * +----+----+       +----+----+
+     * |   Bert  |       |  Chuck  |
+     * | 2     3 |       | 4     11|
+     * +---------+       +----+----+
+     *                        |
+     *         +-------------------------+
+     *         |             |            |
+     *     +----+----+  +----+----+  +----+----+
+     *     |  Donna  |  |  Eddie  |  |   Fred  |
+     *     | 5     6 |  | 7     8 |  | 9     10|
+     *     +---------+  +---------+  +---------+
+     *
+     * @returns
+     */
+    async function createTestTableWithTestData() {
+        const tableName = await createTestTable();
+        await pool.query(`INSERT INTO "${tableName}" ("name", "left", "right") VALUES
+        ('Albert', 1, 12),
+        ('Bert', 2, 3),
+        ('Chuck', 4, 11),
+        ('Donna', 5, 6),
+        ('Eddie', 7, 8),
+        ('Fred', 9, 10)`);
+        return tableName;
+    }
+
+    async function getNodeIdFromName(tableName: string, name: string) {
+        const queryer = new NestedSetModelQueryer(pool, tableName);
+        const nodes = await queryer.getNodesByName(name);
+        if (!nodes) throw new Error(`Can't find node by name: ${name}`);
+        return nodes[0]["id"];
+    }
+
+    it("Test `getNodesByName`", async () => {
```

Comment:
Ideally specs should be written like `$THING should do $BEHAVIOUR`, rather than just "test" - would it be ok to write what the test are looking for in the `it` call?

Reply:
rewrote the test cases desciption